rasterio/_vsiopener.pyx

```diff
@@ -572,6 +593,8 @@ def _create_opener(obj):
     """Adapt Python file and fsspec objects to the opener interface."""
     if isinstance(obj, _AbstractOpener):
         opener = obj
+    elif issubclass(obj, _AbstractOpener):
+        opener = obj()
```

Comment:
This could be removed, but I found that if you passed an class built using `_AbstractOpener` you will then have a `callable` and thus end up at line 599, using `_FileOpener(obj)`, while ideally you want to have `obj()`